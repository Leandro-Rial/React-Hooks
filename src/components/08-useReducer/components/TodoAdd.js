import React from "react";
import useForm from "../../hooks/useForm";

const TodoAdd = ({ handleAddTodo }) => {
  const [{ description }, handleInputChange, reset] = useForm({
    description: "",
  });

  const handleSubmit = (e) => {
    e.preventDefault();

    if (description.trim().length <= 1) {
      return;
    }

    const newTodo = {
      id: new Date().getTime(),
      desc: description,
      done: false,
    };

    handleAddTodo(newTodo);

    reset();
  };

  return (
    <>
      <h1>Agregar todo</h1>
      <hr />

      <form onSubmit={handleSubmit}>
        <input
          type="text"
          name="description"
          placeholder="Aprender..."
          className="form-control"
          value={description}
          onChange={handleInputChange}
          autoComplete="off"
        />

        <button
          type="submit"
          className="btn btn-outline-primary mt-4 btn-block"
        >
          Agregar
        </button>
      </form>
    </>
  );
};

export default TodoAdd;
