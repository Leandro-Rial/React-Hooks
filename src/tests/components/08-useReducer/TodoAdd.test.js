import { shallow } from "enzyme";
import TodoAdd from "../../../components/08-useReducer/components/TodoAdd";

describe("Pruebas en TodoAdd", () => {
  const handleAddTodo = jest.fn();

  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);

  test("debe de mostrar <TodoAdd /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("no debe de llamar handleAddTodo", () => {
    const submit = wrapper.find("form").prop("onSubmit");

    submit({ preventDefault() {} });

    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test("debe de llamar la funcion handleAddTodo", () => {
    const value = "Aprender React";

    wrapper.find("input").simulate("change", {
      target: {
        value,
        name: "description",
      },
    });

    const submit = wrapper.find("form").prop("onSubmit");

    submit({ preventDefault() {} });

    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
    expect(handleAddTodo).toHaveBeenCalledWith({
        id: expect.any(Number),
        desc: value,
        done: false
    });

    expect(wrapper.find('input').prop('value')).toBe('')
  });
});
