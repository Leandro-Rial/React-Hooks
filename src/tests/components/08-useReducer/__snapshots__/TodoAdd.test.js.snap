// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoAdd debe de mostrar <TodoAdd /> correctamente 1`] = `
<Fragment>
  <h1>
    Agregar todo
  </h1>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-4 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
